packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Deputy/JsiiTypeAttributeBase.cs

```diff
@@ -17,27 +20,32 @@ protected JsiiTypeAttributeBase(Type nativeType, string fullyQualifiedName)
             Load(nativeType.Assembly);
         }
 
-        void Load(Assembly assembly)
+        private static void Load(Assembly assembly)
         {
-            IEnumerable<Assembly> dependencies = assembly.GetReferencedAssemblies()
-                .Select(assemblyName => Assembly.Load(assemblyName));
+            if (ProcessedAssemblies.Contains(GetAssemblyKey(assembly)))
```

Comment:
While `ConcurrentBag` is thread-safe, this check isn't and can result in duplicates loads. But if `JsiiTypeAttributeBase.Load` is idempotent and the performance impact of the occasional duplicate load isn't significant, this is good enough. Can you add a comment that mentions the idempotency requirement?

If thread-safety isn't a requirement, there's no need to use `ConcurrentBag`.

Reply:
If each call to js is an independent call, it's possible that someone could write a multithreaded application that works.

I'm not saying they should. But if it doesn't work, I don't want a cache to be the reason.

I don't think there has been a concrete stance on multithreading/async calls. The CDK people are still talking about options as well.